### ISSUE TITLE
manual generation works but demo test does not

### ISSUE THREAD

**Issue by terminills** [terminills/gator#239]
python3 - <<'EOF'
from diffusers import StableDiffusionXLPipeline; import torch
pipe = StableDiffusionXLPipeline.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.float16, variant="fp16", use_safetensors=True).to("cuda") 
print(pipe("a test image", num_inference_steps=5).images[0])
EOF

Loading pipeline components...:  14%|██████████████▍                                                                                      | 1/7 [00:00<00:01,  3.32it/s]`torch_dtype` is deprecated! Use `dtype` instead!
Loading pipeline components...: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████| 7/7 [00:00<00:00,  9.54it/s]
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:05<00:00,  1.13s/it]
<PIL.Image.Image image mode=RGB size=1024x1024 at 0x70EF28566CB0>
git pull
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (5/5), done.
remote: Total 14 (delta 7), reused 12 (delta 7), pack-reused 0 (from 0)
Unpacking objects: 100% (14/14), 8.08 KiB | 393.00 KiB/s, done.
From https://github.com/terminills/gator
   e3dadeb..852e90b  main                                             -> origin/main
 * [new branch]      copilot/fix-content-generation-issue-another-one -> origin/copilot/fix-content-generation-issue-another-one
Updating e3dadeb..852e90b
Fast-forward
 src/backend/services/ai_models.py | 47 +++++++++++++++++++++++++++++++++++------------
 1 file changed, 35 insertions(+), 12 deletions(-)
python ../demo_ai_video_generation.py

======================================================================
🎥 GATOR AI-POWERED VIDEO GENERATION DEMO
Enhanced Video Features with Real Image Generation
======================================================================

======================================================================
🎬 DEMO: Placeholder vs AI-Generated Video Frames
======================================================================

1. Initializing AI Model Manager...
   - GPU Type: rocm
   - GPU Memory: 59.96875 GB
   - Available image models: 3

----------------------------------------------------------------------
2. Generating video with PLACEHOLDER frames (fast mode)
----------------------------------------------------------------------
   Prompts: ['A serene mountain landscape at sunrise', 'A bustling city skyline at night']
   Quality: STANDARD (1280x720)
   Use AI: False

✅ Placeholder video generated!
   📁 Path: generated_content/videos/video_multi_frame_20251112_135356.mp4
   ⏱️  Duration: 4.0 seconds
   📊 File size: 46.08 KB
   🎞️  Scenes: 2

----------------------------------------------------------------------
3. Generating video with AI-GENERATED frames (quality mode)
----------------------------------------------------------------------
   Prompts: ['A serene mountain landscape at sunrise', 'A bustling city skyline at night']
   Quality: STANDARD (1280x720)
   Use AI: True
   Note: AI generation may take several minutes for first run
         (model download + generation time)
Loading pipeline components...:   0%|                                                                                                             | 0/5 [00:00<?, ?it/s]`torch_dtype` is deprecated! Use `dtype` instead!
Loading pipeline components...: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:00<00:00, 11.19it/s]
xformers not available, using default attention
  0%|                                                                                                                                            | 0/20 [00:00<?, ?it/s]
Diffusers generation failed: argument of type 'NoneType' is not iterable
❌ IMAGE GENERATION FAILED after 9.00s
   Error: argument of type 'NoneType' is not iterable
   Model attempted: sdxl-1.0
AI frame generation failed: argument of type 'NoneType' is not iterable, using placeholder
  0%|                                                                                                                                            | 0/20 [00:00<?, ?it/s]
Diffusers generation failed: argument of type 'NoneType' is not iterable
❌ IMAGE GENERATION FAILED after 0.04s
   Error: argument of type 'NoneType' is not iterable
   Model attempted: sdxl-1.0
AI frame generation failed: argument of type 'NoneType' is not iterable, using placeholder

✅ AI-generated video created!
   📁 Path: generated_content/videos/video_multi_frame_20251112_135406.mp4
   ⏱️  Duration: 4.0 seconds
   📊 File size: 46.08 KB
   🎞️  Scenes: 2

   💡 Compare the two videos to see the difference!
      Placeholder: Simple gradients with text
      AI-generated: Actual images from prompts

======================================================================
🎬 DEMO: Direct VideoProcessingService with AI
======================================================================

1. Initializing services...

2. Generating multi-scene video with AI...
   Scenes: 3
   Quality: HIGH (1920x1080)
   Transition: CROSSFADE
Loading pipeline components...: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:00<00:00,  7.20it/s]
xformers not available, using default attention
  0%|                                                                                                                                            | 0/20 [00:00<?, ?it/s]
Diffusers generation failed: argument of type 'NoneType' is not iterable
❌ IMAGE GENERATION FAILED after 5.44s
   Error: argument of type 'NoneType' is not iterable
   Model attempted: sdxl-1.0
AI frame generation failed: argument of type 'NoneType' is not iterable, using placeholder
  0%|                                                                                                                                            | 0/20 [00:00<?, ?it/s]
Diffusers generation failed: argument of type 'NoneType' is not iterable
❌ IMAGE GENERATION FAILED after 0.05s
   Error: argument of type 'NoneType' is not iterable
   Model attempted: sdxl-1.0
AI frame generation failed: argument of type 'NoneType' is not iterable, using placeholder
  0%|                                                                                                                                            | 0/20 [00:00<?, ?it/s]
Diffusers generation failed: argument of type 'NoneType' is not iterable
❌ IMAGE GENERATION FAILED after 0.05s
   Error: argument of type 'NoneType' is not iterable
   Model attempted: sdxl-1.0
AI frame generation failed: argument of type 'NoneType' is not iterable, using placeholder

✅ Video generated with AI frames!
   📁 Path: /tmp/gator_ai_videos/video_multi_frame_20251112_135412.mp4
   ⏱️  Total duration: 6.0 seconds
   📐 Resolution: 1920x1080
   🎞️  Scenes: 3
   🔀 Transitions: crossfade
   📊 File size: 73.11 KB

======================================================================
✅ ALL DEMOS COMPLETED!
======================================================================

📁 Generated videos are in:
   - AI Manager: generated_content/videos/
   - Direct service: /tmp/gator_ai_videos/

📝 Key Points:
   - use_ai_generation=False: Fast placeholder frames
   - use_ai_generation=True: Real AI-generated images
   - Requires GPU or API keys for AI generation
   - Fallback to placeholder if AI generation fails

🧪 Run tests:
   pytest tests/unit/test_ai_video_frame_generation.py -v
